obelisk/python/obelisk_py/node.py

```diff
@@ -1,86 +1,515 @@
+
+        Raises:
+            ValueError: If the configuration string is invalid.
+        """
+        # parse and check the configuration string
+        field_names, value_names = [], []
+        for config_str in config_strs:
+            field_name, value_name = ObeliskNode._parse_config_str(config_str)
+            field_names.extend(field_name)
+            value_names.extend(value_name)
+
+        if (not field_names and not value_names) or config_strs == [""]:
+            return {}  # case: no callback groups to create
+
+        allowable_value_names = ["MutuallyExclusiveCallbackGroup", "ReentrantCallbackGroup"]
+        ObeliskNode._check_values(value_names, allowable_value_names)
+        config_dict = dict(zip(field_names, value_names))
+
+        # create the callback groups
+        callback_group_dict = {}
+        for callback_group_name, callback_group_type in config_dict.items():
+            if callback_group_type == "MutuallyExclusiveCallbackGroup":
+                callback_group = MutuallyExclusiveCallbackGroup()
+            elif callback_group_type == "ReentrantCallbackGroup":
+                callback_group = ReentrantCallbackGroup()
+            else:
+                raise ValueError(f"Invalid callback group type: {callback_group_type}")
+            callback_group_dict[callback_group_name] = callback_group
+
+        return callback_group_dict
+
+    def _create_publisher_from_config_str(self, config_str: str, msg_type: Optional[Type] = None) -> Publisher:
+        """Create a publisher from a configuration string.
+
+        [NOTE] There are many unsupported features in this function, such as setting QoS profiles, callback groups, etc.
+        For now, we assume the only property of the QoS profile the user will set is history depth, and we don't even
+        expose the event_callbacks, qos_overriding_options, or publisher_class parameters.
+
+        Parameters:
+            config_str: The configuration string.
+            msg_type: The message type. If passed, we just use this directly. Otherwise, you can configure this at
+                runtime by passing a msg_type field in the config string that corresponds to an Obelisk message type.
+
+        Returns:
+            publisher: The created publisher.
+
+        Raises:
+            AssertionError: If the configuration string is invalid.
+            ObeliskMsgError: If the message type is not an Obelisk message.
+        """
+        # parse and check the configuration string
+        field_names, value_names = ObeliskNode._parse_config_str(config_str)
+        required_field_names = ["topic"]
+        optional_field_names = ["msg_type", "history_depth", "callback_group", "non_obelisk"]
+        ObeliskNode._check_fields(field_names, required_field_names, optional_field_names)
+        config_dict = dict(zip(field_names, value_names))
+
+        # parse the message type
+        if msg_type is None:
```

Comment:
So there is a message type argument in this function, but if its not provided, you grab it from the config string? Just want to make sure I understand it correctly

Reply:
Yes, that's correct